.github/actions/mint-token/action.yml

```diff
@@ -32,16 +35,29 @@ runs:
         id_token_audience: '${{ inputs.service_audience }}'
         id_token_include_email: true
     - id: 'mint-token'
-      shell: "bash"
       env:
-        token: ${{ steps.auth.outputs.id_token }}
-      run: |
-        OIDC_TOKEN=$(curl -sLS "${ACTIONS_ID_TOKEN_REQUEST_URL}&audience=github-token-minter" -H "User-Agent: actions/oidc-client" -H "Authorization: Bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN");
-        JWT=$(echo $OIDC_TOKEN | jq -j '.value');
-        APP_TOKEN=$(curl -X POST -d'{"repositories":["github-token-minter"],"permissions":{"issues":"read"}}' \
-          -H "Authorization: Bearer ${{env.token}}" \
-          -H "X-GitHub-OIDC-Token: ${JWT}" \
-          ${{ inputs.service_url}}/token);
-        TOKEN=$(echo ${APP_TOKEN} | jq -j '.token');
-        echo "token=$(echo $TOKEN)" >> $GITHUB_OUTPUT;
-
+        TOKEN: '${{ steps.auth.outputs.id_token }}'
+      uses: actions/github-script@v6
+      with:
+        script: |
+          try {
+              const {TOKEN} = process.env;
```

Comment:
nit: do we care about spacing when pulling out variables? i.e. `const { Token } = process.env`

If not feel free to ignore.